### PR TITLE
Do not activate if fetch returns NO_CHANGE

### PR DESCRIPTION
**Pull request by dmandar**
Fixes internal bug: 143622863. We activate only if fetched data has not resulted in a data change. We track this using the last time the ETag was changed (corresponds to a real change to the RC template). 
This change also mirrors current android behavior.

### CODE REVIEW

File: FirebaseRemoteConfig/Sources/Private/RCNConfigSettings.h [hunk 1]
```diff
@@ -76,6 +76,8 @@
 @property(nonatomic, readwrite, assign) NSTimeInterval lastSetDefaultsTimeInterval;
 /// The latest eTag value stored from the last successful response.
 @property(nonatomic, readwrite, assign) NSString *lastETag;
+/// The timestamp of the last eTag update.
+@property(nonatomic, readwrite, assign) NSTimeInterval lastETagUpdateTime;
```
Comment (chliangGoogle) [inline FirebaseRemoteConfig/Sources/Private/RCNConfigSettings.h]: should we persist this value? cuz what happened before the current app launch.

File: FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m [hunk 2]
```diff
@@ -560,6 +560,132 @@ - (void)testFetchConfigsFailed {
...
+    settings.lastETagUpdateTime = 100;
+    settings.lastApplyTimeInterval = 101;
+
+    dispatch_queue_t queue =
+        dispatch_queue_create([[NSString stringWithFormat:@"testNoStatusFetchQueue: %d", i]
+                                  cStringUsingEncoding:NSUTF8StringEncoding],
+                              DISPATCH_QUEUE_SERIAL);
+    _configFetch[i] = OCMPartialMock([[RCNConfigFetch alloc] initWithContent:configContent
+                                                                   DBManager:_DBManager
+                                                                    settings:settings
```
Comment (chliangGoogle) [inline FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m]: is this format right? Maybe run script/style.sh

File: FirebaseRemoteConfig/Sources/Private/RCNConfigSettings.h (hunk 1 above)
Comment (dmandar) [reply FirebaseRemoteConfig/Sources/Private/RCNConfigSettings.h]: good pt. Done.

File: FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m (hunk 2 above)
Comment (dmandar) [reply FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m]: This is with style.sh.

**Pull request merged**